mlir/lib/Dialect/LLVMIR/IR/LLVMDialect.cpp

```diff
@@ -2848,6 +2848,41 @@ struct LLVMOpAsmDialectInterface : public OpAsmDialectInterface {
 // DialectInlinerInterface
 //===----------------------------------------------------------------------===//
 
+[[nodiscard]] static bool allocaHasConstantSize(LLVM::AllocaOp allocaOp) {
+  Operation *definingOp = allocaOp.getArraySize().getDefiningOp();
+  return definingOp && definingOp->hasTrait<OpTrait::ConstantLike>();
```

Comment:
Could this be a use case for a matcher from the tutorial? Somthing with matchPattern / m_Constant.

Reply:
I'm not sure what you mean? 🤔